app/controllers/api/users.php

```diff
@@ -1080,6 +1080,61 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
         $response->dynamic(new Document($prefs), Response::MODEL_PREFERENCES);
     });
 
+App::post('/v1/users/:userId/tokens')
+    ->desc('Create session token')
```

Comment:
What's the use case for this again? I can see this being used for [full server-side impersonation](https://github.com/appwrite/appwrite/issues/3891), but it would require an additional API call to exchange the token for a session.

Reply:
Mainly for using a custom messaging method for authentication.
For example, if you want to implement login via WhatsApp link.
Server side impersonation is also a good use case.

Reply:
An additional API call for the exchange when doing full server-side impersonation might be a pain, though.